### ISSUE TITLE
Setting custom LB name

### ISSUE THREAD

**Issue by d-cs** [kubernetes-sigs/aws-load-balancer-controller#1074]
Great project. Very much enjoy using it!

Is it possible to set the Load Balancers `Name` as it appears in the AWS Console as part of the manifests? We produce several of these LB's using Helm charts, the ingress names and namespaces are identical in each cluster, making it inconvenient to determine which LB is on which cluster. We're currently using the tags to find the right cluster but it would make things marginally easier.

**Comment by kerma** [comment]
Currently this is not possible. There is a controller parameter where you can set alb prefix: https://github.com/kubernetes-sigs/aws-alb-ingress-controller/blob/e5c5d6e293e2b81167791ca9b2cea39fc5af7078/internal/ingress/controller/config/config.go#L75

..but it's not really useful in case where you need to find the ALB programmatically.

I'm thinking of trying to take a stab at this, as the workarounds are not that nice. Would be good to get a confirmation from the maintainers that they'd accept the PR.

**Comment by leoskyrocker** [comment]
@M00nF1sh could you kindly chip in your opinion?

**Comment by leoskyrocker** [comment]
@M00nF1sh could you kindly chime in?

**Comment by jescarri** [comment]
I've been wondering about this too, is there a way the controller can write to the ingress status:
1.- ALB name.
2.- ALB arn.

maybe target group arn?

It will improve the UX, in our case we have large multi tenant clusters, our users don't have access to the AWS accounts, every time they need the ALB name or ARN they have to contact us.

If this is a desirable approach I might be able to submit a PR if @M00nF1sh doesn't beat me to :).

Thanks!

**Comment by leoskyrocker** [comment]
IMO it could be as simple as removing the prefix. The prefix is already an empty string by default. I think the code should avoid generating a prefix if it is by default not provided.
With "namespace" + "ingressName" + md5sum, it is already quite sufficient to avoid collision. Though I assume the reason to have a prefix is exactly to avoid a collision.. so maybe we will need to have some collision handling code.

**Comment by omnibs** [comment]
This also has the potential to make consuming ALB logs and metrics way easier.

The whole problem boils down to a scenario that might be pretty common:
* One AWS account
* Multiple clusters per environment (staging, production, development)
* Same namespace and ingress name in each cluster
* Folks struggling to figure out what cluster's a load balancer tied to. Including all the resources related to it (metrics, logs).

**Comment by M00nF1sh** [comment]
Hi guys,
with the new v2.0.0 controller, we have applied AWS tags to resources, which will allow to track resources.
Also, we removed support for name prefix and used a general pattern for load balancer name: "k8s-namespace-ingName-hash".
We'll add support to customize ALB name soon, tracking it at https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/1483

@jescarri we'll add ALB ARN into Ingress annotations, tracking it at https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/1573

closing this issue.
